lib/generators/debugs_bunny/trace/trace_generator.rb

```diff
@@ -3,14 +3,15 @@
 require 'rails/generators'
 
 module DebugsBunny
-  class ModelGenerator < Rails::Generators::NamedBase
-    source_root File.expand_path('templates', __dir__)
+  class TraceGenerator < Rails::Generators::NamedBase
+    TEMPLATE_DIR = File.join(TEMPLATES_DIR, 'model')
+    source_root TEMPLATE_DIR
 
     def generate_model
       models_dir = File.join('app', 'models')
       model_file = File.join(models_dir, "#{file_name}.rb")
```

Comment:
Where is `file_name` coming from?

Reply:
`file_name` is an inherited `attr_reader` from `NamedBase`. When invoking a `NamedBase` generator, a mandatory arg is the name of the thing being generated. In this case, we invoke it with `rails g debugs_bunny:trace DebugTrace`, and it will generate a file called `debug_trace.rb`